sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClient.java

```diff
@@ -110,7 +229,23 @@ public TableServiceVersion getServiceVersion() {
      * {@link AzureNamedKeyCredential}.
      */
     public String generateSas(TableSasSignatureValues tableSasSignatureValues) {
-        return client.generateSas(tableSasSignatureValues);
+        AzureNamedKeyCredential azureNamedKeyCredential = TableSasUtils.extractNamedKeyCredential(getHttpPipeline());
+
+        if (azureNamedKeyCredential == null) {
+            throw logger.logExceptionAsError(new IllegalStateException("Cannot generate a SAS token with a client that"
+                + " is not authenticated with an AzureNamedKeyCredential."));
+        }
+
+        return new TableSasGenerator(tableSasSignatureValues, getTableName(), azureNamedKeyCredential).getSas();
+    }
+
+
+    private Context setContext(Context context) {
+        return context == null ? Context.NONE : context;
+    }
+
+    private Long setTimeout(Duration timeout) {
+        return timeout != null ? timeout.toMillis() : Duration.ofDays(1).toMillis();
```

Comment:
Can be moved to common util?

Reply:
[Ditto question here](https://github.com/Azure/azure-sdk-for-java/pull/33143#discussion_r1084439563)